x/observer/keeper/msg_server_add_observer.go

```diff
@@ -0,0 +1,54 @@
+package keeper
+
+import (
+	"context"
+	"math"
+
+	cosmoserrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func (k msgServer) AddObserver(goCtx context.Context, msg *types.MsgAddObserver) (*types.MsgAddObserverResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if msg.Creator != k.GetParams(ctx).GetAdminPolicyAccount(types.Policy_Type_group2) {
```

Comment:
Can we document it here? https://www.notion.so/zetachain/Observer-Policy-Groups-4e501dd64dfc460180a361f9b8507310?pvs=4